rfcs/high-contrast-theme.md

```diff
@@ -0,0 +1,125 @@
+
+```css
+@media (forced-colors: active) {
+  /* HC is enabled in Windows */
+
+  @media (prefers-color-scheme: light) {
+    /* The HC theme is dark on light */
+  }
+
+  @media (prefers-color-scheme: dark) {
+    /* The HC theme is light on dark */
+  }
+}
+```
+
+More details are available in [Edge blog: Styling for Windows high contrast with new standards for forced colors](https://blogs.windows.com/msedgedev/2020/09/17/styling-for-windows-high-contrast-with-new-standards-for-forced-colors/).
+
+Besides the media query, [CSS system colors](https://www.w3.org/TR/css-color-4/#css-system-colors) are set to match the colors defined by the HC theme. User can customize following colors which are then used to theme the UI:
+
+- `CanvasText`
+- `Canvas`
+- `LinkText`
+- `GrayText`
+- `HighlightText`
+- `Highlight`
+- `ButtonText`
+- `ButtonFace`
+
+#### High Contrast Black (Windows built-in, matches FUI HC theme)
+
+![Black High contrast theme screenshot](high-contrast-theme-black.png)
+
+#### High Contrast White (Windows built-in)
+
+![White High contrast theme screenshot](high-contrast-theme-white.png)
+
+#### High Contrast with custom user-defined colors
+
+![Custom High contrast theme screenshot](high-contrast-theme-custom.png)
+
+The screenshots are from a [codesandbox showing the usage of CSS system colors](https://codesandbox.io/s/high-contrast-1usny?file=/index.html) to render the same layout as Windows settings dialog for HC.
+
+### Implementation in FUI theme
+
+Current HC FUI theme uses the following five colors - `white`, `black`, `hyperlink`, `disabled`, `selected`:
+![Current HC colors](high-contrast-theme-current-colors.png).
+
+A simple solution would be to create five new color tokens and, using a combination of `forced-colors` media query and css variables, map them to the FUI colors if `forced-colors` are not active and to system colors if `forced-colors` are active.
+
+#### Mapping table
+
+| Token        | FUI color | System color |
+| ------------ | --------- | ------------ |
+| hcCanvas     | black     | Canvas       |
+| hcCanvasText | white     | CanvasText   |
+| hcHyperlink  | hyperlink | LinkText     |
+| hcDisabled   | disabled  | GrayText     |
+| hcSelected   | selected  | Highlight    |
+
+#### Example code
+
+```css
+/* FUI colors */
+:root {
+  --global-color-hc-canvas: var(--global-color-black); /* #000000 */
+  --global-color-hc-canvas-text: var(--global-color-white); /* #FFFFFF */
+  --global-color-hc-hyperlink: var(--global-color-hyperlink); /* #FFFF00 */
+  --global-color-hc-disabled: var(--global-color-selected); /* #1AEBFF */
+  --global-color-hc-selected: var(--global-color-disabled); /* #3FF23F */
+}
+
+@media (forced-colors: active) {
+  /* System colors */
+  :root {
+    --global-color-hc-canvas: Canvas;
+    --global-color-hc-canvas-text: CanvasText;
+    --global-color-hc-hyperlink: LinkText;
+    --global-color-hc-disabled: GrayText;
+    --global-color-hc-selected: Highlight;
+  }
+}
+```
+
```

Comment:
depending on if we implement system colors for `react` and `react-northstar` we might need to ship a black HC theme as well - due to compatibility reasons. It might of course be easier to just implement system colors in both of these flavors of Fluent UI, but we need to make a decision on that and understand implications for the consumers.